observability/metrics/src/test/lib/scala/com/daml/metrics/api/testing/InMemoryMetricsFactory.scala

```diff
@@ -165,15 +168,29 @@ object InMemoryMetricsFactory extends InMemoryMetricsFactory {
 
   case class InMemoryGauge[T](context: MetricsContext, initial: T) extends Gauge[T] {
     val value = new AtomicReference[T](initial)
+    val closed = new AtomicBoolean(false)
 
     override def name: String = "test"
 
-    override def updateValue(newValue: T): Unit =
+    override def updateValue(newValue: T): Unit = {
+      checkClosed()
       value.set(newValue)
+    }
+
+    override def updateValue(f: T => T): Unit = {
+      checkClosed()
+      discard(value.updateAndGet(value => f(value)))
+    }
+
+    override def getValue: T = {
+      checkClosed()
+      value.get()
+    }
 
-    override def getValue: T = value.get()
+    override def close(): Unit = closed.set(true)
 
-    override def updateValue(f: T => T): Unit = discard(value.updateAndGet((t: T) => f(t)))
+    private def checkClosed(): Unit =
+      if (closed.get()) throw new IllegalArgumentException("Already closed")
```

Comment:
Wouldn't an `IllegalStateException` be more appropriate here? I.e., it is inappropriate to call `getValue` or `updateValue` on a closed `Gauge`.

Reply:
Good point